### PR TITLE
Java 11 support - Integration Branch

### PR DESCRIPTION
**Pull request by oleg-nenashev**
WiP Branch for Java 11 support.
Java 10 is YAGNI, I believe

@jenkinsci/sig-platform 

Note: this PR also serves as an easy way to see the diff between `master` and the `java11-support` branches.

### CODE REVIEW

**Comment by slide** [comment]
👍

**Review by MarkEWaite — COMMENTED** [review]
Looks good to me for continuing development

**Review by MarkEWaite — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
@jenkinsci/java11-support

**Review by kuisathaverat — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -17,10 +17,11 @@ COPY show-pom-version.rb /jenkins/src/show-pom-version.rb
 WORKDIR /jenkins/src/
 RUN mvn clean install --batch-mode -Plight-test
 
-# The image is based on the previous weekly, new changes in jenkinci/docker are not applied
-FROM jenkins/jenkins:latest
+# The image is based on https://github.com/jenkinsci/docker/tree/java11
+# All documentation is applicable
+FROM jenkins/jenkins-experimental:2.138.1-jdk11
```
Comment (oleg-nenashev) [inline Dockerfile]: Should be replaced by the common release once https://github.com/jenkinsci/docker/pull/762 from @batmat lands in the weekly release

Comment (batmat) [reply Dockerfile]: @oleg-nenashev `docker pull jenkins/jenkins:2.152-jdk11` now is released.

**Review by batmat — APPROVED** [review]

**Review by alecharp — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
This branch is no longer active. All code has been integrated into the master branch, https://issues.jenkins-ci.org/browse/INFRA-1694 is a follow-up for continuous Delivery of Docker images (to be done in a new way after JEP-217)

**Pull request closed without merging**